src/rustdoc-json-types/lib.rs

```diff
@@ -195,13 +195,56 @@ pub struct Item {
     /// - `#[repr(C)]` and other reprs also appear as themselves,
     ///   though potentially with a different order: e.g. `repr(i8, C)` may become `repr(C, i8)`.
     ///   Multiple repr attributes on the same item may be combined into an equivalent single attr.
-    pub attrs: Vec<String>,
+    pub attrs: Vec<Attribute>,
     /// Information about the item’s deprecation, if present.
     pub deprecation: Option<Deprecation>,
     /// The type-specific fields describing this item.
     pub inner: ItemEnum,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+/// An attribute, eg `#[repr(C)]`
+///
+/// This doesn't include:
+/// - `#[doc = "Doc Comment"]` or `/// Doc comment`. These are in [`Item::docs`] instead.
+/// - `#[depricate]`. These are in [`Item::deprecation`] instead.
+pub enum Attribute {
```

Comment:
What is the reason this type is duplicated from `AttributeKind` in the new structured AST?
Having everything duplicated is not great, but if you want this for stability purposes I understand.

Love this PR btw, the structured attributes are great to have